soc/riscv/riscv-privileged/common/soc_common.h

```diff
@@ -41,6 +41,59 @@
 void soc_interrupt_init(void);
 #endif
 
+#if defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
+
+/**
+ * @brief Enable an interrupt line for RISC-V custom interrupt controller.
+ *
+ * This routine enables an interrupt line for RISC-V custom interrupt controller.
+ * Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number to enable
+ */
+void z_soc_irq_enable(uint32_t irq);
```

Comment:
Why are you using `uint32_t` instead of `unsigned int` like everywhere else in the code?